.github/workflows/build-python.yml

```
@@ -42,14 +43,14 @@ jobs:
           ssh-add - <<< "${{ secrets.PYTHON_PRIVATE_REPO_KEY }}"
           git config --global url."git@github.com:toggleglobal".insteadOf https://github.com/toggleglobal
 
-      - name: Install dependencies (Pip)
+      - name: Dependencies - Pip
```

Try to name steps describing what they do rather than just blunt titles.
```suggestion
      - name: Add Pip Dependencies
```

Similar for poetry